erizo/src/erizo/bandwidth/TargetVideoBWDistributor.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+ * TargetVideoBWDistributor.cpp
+ */
+
+#include <algorithm>
+
+#include "TargetVideoBWDistributor.h"
+#include "MediaStream.h"
+#include "Transport.h"
+#include "rtp/RtpUtils.h"
+
+namespace erizo {
+
+void TargetVideoBWDistributor::distribute(uint32_t remb, uint32_t ssrc,
+                                  std::vector<std::shared_ptr<MediaStream>> streams, Transport *transport) {
+  std::sort(streams.begin(), streams.end(),
+    [this](const std::shared_ptr<MediaStream> &i, const std::shared_ptr<MediaStream> &j) {
+      return getTargetVideoBW(i) < getTargetVideoBW(j);
+    });
+  uint8_t remaining_streams = streams.size();
+  uint32_t remaining_bitrate = remb;
+  std::for_each(streams.begin(), streams.end(),
+    [&remaining_bitrate, &remaining_streams, transport, ssrc, this](const std::shared_ptr<MediaStream> &stream) {
+      uint32_t max_bitrate = stream->getMaxVideoBW();
+
+      uint32_t target_bitrate = getTargetVideoBW(stream);
+
+      uint32_t remaining_avg_bitrate = remaining_bitrate / remaining_streams;
+      uint32_t bitrate = std::min(target_bitrate, remaining_avg_bitrate);
+      uint32_t remb = std::min(max_bitrate, remaining_avg_bitrate);
+      uint32_t bitrate_to_transfer = std::max(bitrate - target_bitrate * 1.2, 0.0);
```

Comment:
I don't think we need `bitrate_to_transfer` it will always be 0 unless we have a bug somewhere else. As far as I can tell `bitrate` is enough to update `remaining_bitrate` and distribute the bitrate correctly.

Reply:
good catch!